Framework/Geometry/src/Instrument/InstrumentDefinitionParser.cpp

```diff
@@ -2272,18 +2288,50 @@ void InstrumentDefinitionParser::setLogfile(
       continue;
     }
 
-    // if more than one <value> specified for a parameter use only the first
-    // <value> element
+    DateAndTime currentValidFrom;
+    DateAndTime currentValidTo;
+    currentValidFrom.setToMinimum();
+    currentValidTo.setToMaximum();
+
+    // if more than one <value> specified for a parameter, check the validity
+    // range
     if (numberValueEle >= 1) {
-      pValueElem = static_cast<Element *>(pNLvalue->item(0));
-      if (!pValueElem->hasAttribute("val"))
+      bool hasValue = false;
+
+      for (unsigned long i = 0; i < numberValueEle; i++) {
+        pValueElem = static_cast<Element *>(pNLvalue->item(i));
+
+        if (!pValueElem->hasAttribute(("val")))
+          continue;
+
+        validFrom.setToMinimum();
+        if (pValueElem->hasAttribute("valid-from"))
+          validFrom.setFromISO8601(pValueElem->getAttribute("valid-from"));
```

Comment:
again, what if it is not in the right format? will it throw?

Reply:
Yes, it throws an error